x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/helpers.ts

```diff
@@ -94,27 +100,57 @@ export const streamGraph = async ({
       const event = value;
       // only process events that are part of the agent run
       if ((event.tags || []).includes(AGENT_NODE_TAG)) {
-        if (event.event === 'on_llm_stream') {
-          const chunk = event.data?.chunk;
-          // TODO: For Bedrock streaming support, override `handleLLMNewToken` in callbacks,
-          // TODO: or maybe we can update ActionsClientSimpleChatModel to handle this `on_llm_stream` event
-          if (event.name === 'ActionsClientChatOpenAI') {
+        if (event.name === 'ActionsClientChatOpenAI') {
+          if (event.event === 'on_llm_stream') {
+            const chunk = event.data?.chunk;
             const msg = chunk.message;
-
-            if (msg.tool_call_chunks && msg.tool_call_chunks.length > 0) {
+            if (msg?.tool_call_chunks && msg?.tool_call_chunks.length > 0) {
+              // I don't think we hit this anymore because of our check for AGENT_NODE_TAG
+              // however, no harm to keep it in
```

Comment:
👍 I don't think so as well. Can remove once we finish these last remaining graph changes -- I know @YulNaumenko has some smaller graph changes as part of https://github.com/elastic/kibana/pull/188172